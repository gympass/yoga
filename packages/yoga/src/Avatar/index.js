import Avatar from './web/Avatar';
import AvatarCircle from './web/AvatarCircle';

const ExportAvatar = Object.assign(Avatar, { Circle: AvatarCircle });

export default ExportAvatar;
