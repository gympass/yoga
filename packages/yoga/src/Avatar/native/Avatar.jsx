import React, { forwardRef } from 'react';
import { BuildingFilled } from '@gympass/yoga-icons';
import { string, func, number } from 'prop-types';
import { ImagePropTypes } from 'deprecated-react-native-prop-types';
import { Image } from 'react-native';

import styled from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';
import Box from '../../Box';
import Icon from '../../Icon';

const BORDER_WIDTH = 1;
const BORDER_OPACITY = 0.1;

const BorderImage = styled(Box).attrs(({ theme: { yoga } }) => {
  return {
    borderColor: hexToRgb(yoga.colors.secondary, BORDER_OPACITY),
    borderWidth: BORDER_WIDTH,
  };
})`
  position: absolute;
`;

const Content = ({ icon, src, fill, content, stroke }) => {
  if (src) {
    return <Box as={Image} width="100%" height="100%" source={src} />;
  }

  if (content) {
    return content;
  }

  return (
    <Icon as={icon} width="50%" height="50%" fill={fill} stroke={stroke} />
  );
};

/**
 * The Avatar component is used to display the image.
 * It has two shapes: default and circle
 */
const Avatar = forwardRef(
  (
    {
      icon = BuildingFilled,
      src,
      children,
      fill,
      stroke,
      borderRadius,
      width,
      height,
      ...props
    },
    ref,
  ) => {
    return (
      <Box
        ref={ref}
        bgColor="elements.selectionAndIcons"
        display="flex"
        alignItems="center"
        justifyContent="center"
        width={width}
        height={height}
        overflow="hidden"
        borderRadius={borderRadius}
        {...props}
      >
        <Content
          icon={icon}
          src={src}
          fill={fill}
          stroke={stroke}
          content={children}
        />
        <BorderImage width="100%" height="100%" borderRadius={borderRadius} />
      </Box>
    );
  },
);

Content.propTypes = {
  src: ImagePropTypes.source,
  icon: func,
  fill: string,
  ...Box.propTypes,
};

Content.defaultProps = {
  src: undefined,
  fill: 'white',
  icon: BuildingFilled,
};

Avatar.propTypes = {
  src: ImagePropTypes.source,
  icon: func,
  fill: string,
  width: number,
  height: number,
  borderRadius: string,
  ...Box.propTypes,
};

Avatar.defaultProps = {
  src: undefined,
  fill: 'white',
  width: 48,
  height: 48,
  borderRadius: 'small',
  ...Box.defaultProps,
};

Avatar.displayName = 'Avatar';

export default Avatar;
