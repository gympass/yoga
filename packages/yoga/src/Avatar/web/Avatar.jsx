import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { BuildingFilled } from '@gympass/yoga-icons';
import { string, func, checkPropTypes } from 'prop-types';

import Box from '../../Box';
import Icon from '../../Icon';

const Image = styled.img`
  max-width: 100%;
`;

const Content = ({ icon, alt, src, fill, content, stroke }) => {
  if (src) {
    return <Image src={src} alt={alt} />;
  }

  if (content) {
    return content;
  }

  return (
    <Icon as={icon} width="50%" height="50%" fill={fill} stroke={stroke} />
  );
};

/**
 * The Avatar component is used to display the image.
 * It has two shapes: default and circle
 */
const Avatar = forwardRef(
  (
    {
      src,
      alt,
      fill = 'white',
      stroke,
      icon = BuildingFilled,
      children,
      ...otherProps
    },
    ref,
  ) => {
    return (
      <Box
        bgColor="elements.selectionAndIcons"
        display="flex"
        alignItems="center"
        justifyContent="center"
        width={48}
        height={48}
        borderRadius="small"
        elevation="small"
        overflow="hidden"
        ref={ref}
        {...otherProps}
      >
        <Content
          icon={icon}
          src={src}
          fill={fill}
          stroke={stroke}
          content={children}
          alt={alt}
        />
      </Box>
    );
  },
);

Content.propTypes = {
  src: string,
  icon: func,
  fill: string,
  alt: string,
  ...Box.propTypes,
};

Avatar.propTypes = {
  src: string,
  /**  Required if src is true */
  alt: (props, propName, componentName) => {
    const { src } = props;

    if (src) {
      checkPropTypes(
        { [propName]: string.isRequired },
        props,
        'prop',
        componentName,
      );
    }

    return null;
  },
  /** Used when src is missing */
  icon: (props, propName, componentName) => {
    const { src, [propName]: iconProp } = props;

    if (src && iconProp) {
      return new Error(
        `You should provide a src or ${propName} prop, not both, in '${componentName}'.`,
      );
    }

    checkPropTypes({ [propName]: func }, props, 'prop', componentName);

    return null;
  },
  fill: string,
  stroke: string,
  ...Box.propTypes,
};

Avatar.displayName = 'Avatar';

export default Avatar;
