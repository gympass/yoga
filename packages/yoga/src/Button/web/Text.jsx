import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';
import { bool } from 'prop-types';

import Button from './Button';

const StyledButton = styled(Button)`
  ${({
    secondary,
    inverted,
    theme: {
      yoga: {
        colors,
        components: { button },
      },
    },
  }) => {
    const state = secondary ? 'secondary' : 'primary';

    return `
      background-color: ${button.types.text.backgroundColor};
      border-color: ${button.types.text.backgroundColor};
      color: ${colors[state]};
      text-decoration: underline;
      text-decoration-color: ${button.types.link.font[state].color};

      svg {
        fill: ${colors[state]};
      }

      &:not([disabled]):link {
        text-decoration: underline;
      }

      &:not([disabled]):hover, &:not([disabled]):focus, &:not([disabled]):active {
        background-color: ${button.types.text.backgroundColor};
        box-shadow: none;
      }

      &:not([disabled]):hover {
        color: ${hexToRgb(colors[state], 0.5)};

        text-decoration-color: ${hexToRgb(colors[state], 0.5)};

        svg {
          fill: ${hexToRgb(colors[state], 0.5)};
        }
      }

      &:not([disabled]):focus, &:not([disabled]):active {
        color: ${hexToRgb(colors[state], 0.75)};

        text-decoration-color: ${hexToRgb(colors[state], 0.75)};

        svg {
          fill: ${hexToRgb(colors[state], 0.75)};
        }
      }

      ${
        inverted
          ? `
          color: ${colors.white};

          text-decoration-color: ${colors.white};

          svg {
            fill: ${colors.white};
          }

          &:not([disabled]):hover {
            color: ${hexToRgb(colors.white, 0.5)};

            text-decoration-color: ${hexToRgb(colors.white, 0.5)};

            svg {
              fill: ${hexToRgb(colors.white, 0.5)};
            }
          }

          &:not([disabled]):focus, &:not([disabled]):active {
            color: ${hexToRgb(colors.white, 0.75)};

            text-decoration-color: ${hexToRgb(colors.white, 0.75)};

            svg {
              fill: ${hexToRgb(colors.white, 0.75)};
            }
          }
        `
          : ''
      }

      &:disabled {
        background-color: ${button.types.text.backgroundColor};
        border-color: ${button.types.text.backgroundColor};
        color: ${button.types.text.disabled};

        text-decoration: underline;
        text-decoration-color: ${button.types.text.disabled};

        svg {
          fill: ${button.types.text.disabled};
        }
      }
    `;
  }}
`;

const ButtonText = forwardRef(({ ...rest }, ref) => {
  const props = Object.fromEntries(
    Object.entries(rest).filter(([key]) => key !== 'isLoading'),
  );

  const { secondary = false, inverted = false } = props;

  return (
    <StyledButton
      {...props}
      secondary={secondary}
      inverted={inverted}
      ref={ref}
    />
  );
});

ButtonText.propTypes = {
  inverted: bool,
  secondary: bool,
};

ButtonText.displayName = 'Button.Text';

export default ButtonText;
