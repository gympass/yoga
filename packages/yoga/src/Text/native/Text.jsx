import React from 'react';
import styled from 'styled-components';
import { oneOf, bool } from 'prop-types';
import textStyle from '../textStyle';

const styledText = type => styled.Text`
  ${textStyle(type)}
`;

const H1 = styledText('h1');
H1.displayName = 'Text.H1';

const H2 = styledText('h2');
H2.displayName = 'Text.H2';

const H3 = styledText('h3');
H3.displayName = 'Text.H3';

const H4 = styledText('h4');
H4.displayName = 'Text.H4';

const H5 = styledText('h5');
H5.displayName = 'Text.H5';

const Small = styledText('small');
Small.displayName = 'Text.Small';

const Tiny = styledText('tiny');
Tiny.displayName = 'Text.Tiny';

const Light = styledText('light');
Light.displayName = 'Text.Light';

const Regular = styledText('regular');
Regular.displayName = 'Text.Regular';

const Medium = styledText('medium');
Medium.displayName = 'Text.Medium';

const Bold = styledText('bold');
Bold.displayName = 'Text.Bold';

const Black = styledText('black');
Black.displayName = 'Text.Black';

const SectionTitle = styledText('sectionTitle');
SectionTitle.displayName = 'Text.SectionTitle';

const TextRenderer = styledText('p');

const Text = props => <TextRenderer {...props} />;
Text.displayName = 'Text';

Text.propTypes = {
  inverted: bool,
  /** style the text following the theme */
  variant: oneOf([
    'primary',
    'secondary',
    'vibin',
    'hope',
    'energy',
    'relax',
    'peace',
    'verve',
    'uplift',
    'deepPurple',
    'stamina',
    'deep',
    'medium',
    'light',
    'clear',
    'white',
  ]),
  /** set the font-size following the theme */
  size: oneOf([
    'xxsmall',
    'xsmall',
    'small',
    'medium',
    'large',
    'xlarge',
    'xxlarge',
    'xxxlarge',
    'huge',
  ]),
  /** set the font-weight to regular */
  light: bool,
};

Text.defaultProps = {
  inverted: false,
  variant: undefined,
  size: 'medium',
  light: false,
};

export {
  Text,
  H1,
  H2,
  H3,
  H4,
  H5,
  Small,
  Tiny,
  Light,
  Regular,
  Medium,
  Bold,
  Black,
  SectionTitle,
};
