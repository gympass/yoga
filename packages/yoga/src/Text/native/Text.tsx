import React from 'react';
import styled from 'styled-components/native';
import { oneOf, bool } from 'prop-types';
import { system } from '@gympass/yoga-system';
// @ts-ignore TODO: fix textStyle typing
import textStyle from '../textStyle';
import { deprecated } from '../../shared';

// TODO: Use system typing
const styledText = type =>
  styled.Text<{ children: React.ReactNode; [key: string]: any }>`
    ${textStyle(type)}
    ${system}
  `;

const H1 = styledText('h1');

H1.displayName = 'Text.H1';

const H2 = styledText('h2');

H2.displayName = 'Text.H2';

const H3 = styledText('h3');

H3.displayName = 'Text.H3';

const H4 = styledText('h4');

H4.displayName = 'Text.H4';

const H5 = styledText('h5');

H5.displayName = 'Text.H5';

const Small = styledText('small');

Small.displayName = 'Text.Small';

const Tiny = styledText('tiny');

Tiny.displayName = 'Text.Tiny';

const Light = styledText('light');

Light.displayName = 'Text.Light';

const Regular = styledText('regular');

Regular.displayName = 'Text.Regular';

const Medium = styledText('medium');

Medium.displayName = 'Text.Medium';

const Bold = styledText('bold');

Bold.displayName = 'Text.Bold';

const Black = styledText('black');

Black.displayName = 'Text.Black';

const SectionTitle = styledText('sectionTitle');

SectionTitle.displayName = 'Text.SectionTitle';

const SmallestException = styledText('smallestException');

SmallestException.displayName = 'Text.SmallestException';

const TextRenderer = styledText('p');

const Text = Object.assign(TextRenderer, {
  H1,
  H2,
  H3,
  H4,
  H5,
  Small,
  Tiny,
  Light,
  Regular,
  Medium,
  Bold,
  Black,
  SectionTitle,
  SmallestException,
});

Text.displayName = 'Text';

const fontSizes = oneOf([
  'xxsmall',
  'xsmall',
  'small',
  'medium',
  'large',
  'xlarge',
  'xxlarge',
  'xxxlarge',
  'huge',
]);

Text.propTypes = {
  inverted: bool,
  /** (deprecated: use color instead) style the text following the theme */
  variant: deprecated(
    oneOf([
      'primary',
      'secondary',
      'vibin',
      'hope',
      'energy',
      'relax',
      'peace',
      'verve',
      'uplift',
      'deepPurple',
      'stamina',
      'deep',
      'medium',
      'light',
      'clear',
      'white',
    ]),
    'Use `color` system prop instead',
  ),
  /** (deprecated: use fontSize instead) style the text following the theme */
  size: deprecated(fontSizes, 'Use `fontSize` system prop instead'),
  /** set the font-weight to regular */
  light: bool,
  fontSize: fontSizes,
};

Text.defaultProps = {
  inverted: false,
  variant: undefined,
  size: undefined,
  fontSize: 'medium',
  light: false,
};

export {
  Text,
  H1,
  H2,
  H3,
  H4,
  H5,
  Small,
  Tiny,
  Light,
  Regular,
  Medium,
  Bold,
  Black,
  SectionTitle,
  SmallestException,
};
