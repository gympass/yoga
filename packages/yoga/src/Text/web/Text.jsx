import React from 'react';
import styled from 'styled-components';
import { oneOf, bool, number } from 'prop-types';
import { system } from '@gympass/yoga-system';
import textStyle from '../textStyle.web';
import { deprecated } from '../../shared';

const styledText = (type, element = false) => (element
  ? styled[type]
  : styled.p)`
    margin: 0;
    padding: 0;
    ${textStyle(type)}
    ${system}
`;

const H1 = styledText('h1', true);

H1.displayName = 'Text.H1';

const H2 = styledText('h2', true);

H2.displayName = 'Text.H2';

const H3 = styledText('h3', true);

H3.displayName = 'Text.H3';

const H4 = styledText('h4', true);

H4.displayName = 'Text.H4';

const H5 = styledText('h5', true);

H5.displayName = 'Text.H5';

const Small = styledText('small');

Small.displayName = 'Text.Small';

const Tiny = styledText('tiny');

Tiny.displayName = 'Text.Tiny';

const Light = styledText('light');

Light.displayName = 'Text.Light';

const Regular = styledText('regular');

Regular.displayName = 'Text.Regular';

const Medium = styledText('medium');

Medium.displayName = 'Text.Medium';

const Bold = styledText('bold');

Bold.displayName = 'Text.Bold';

const Black = styledText('black');

Black.displayName = 'Text.Black';

const SectionTitle = styledText('sectionTitle');

SectionTitle.displayName = 'Text.SectionTitle';

const SmallestException = styledText('smallestException');

SmallestException.displayName = 'Text.SmallestException';

const Display1 = styledText('display1');

Display1.displayName = 'Text.Display1';

const Display2 = styledText('display2');

Display2.displayName = 'Text.Display2';

const Display3 = styledText('display3');

Display3.displayName = 'Text.Display3';

const Display4 = styledText('display4');

Display4.displayName = 'Text.Display4';

const DisplayNumber = styledText('displayNumber');

DisplayNumber.displayName = 'Text.DisplayNumber';

const TextRenderer = styledText('p');

const Text = props => <TextRenderer {...props} />;

Text.displayName = 'Text';

Text.H1 = H1;
Text.H2 = H2;
Text.H3 = H3;
Text.H4 = H4;
Text.H5 = H5;
Text.Small = Small;
Text.Tiny = Tiny;
Text.Light = Light;
Text.Regular = Regular;
Text.Medium = Medium;
Text.Bold = Bold;
Text.Black = Black;
Text.SectionTitle = SectionTitle;
Text.SmallestException = SmallestException;
Text.Display1 = Display1;
Text.Display2 = Display2;
Text.Display3 = Display3;
Text.Display4 = Display4;
Text.DisplayNumber = DisplayNumber;

const fontSizes = oneOf([
  'xxsmall',
  'xsmall',
  'small',
  'medium',
  'large',
  'xlarge',
  'xxlarge',
  'xxxlarge',
  'huge',
]);

Text.propTypes = {
  inverted: bool,
  numberOfLines: number,
  /** (deprecated: use color instead) style the text following the theme */
  variant: deprecated(
    oneOf([
      'primary',
      'secondary',
      'vibin',
      'hope',
      'energy',
      'relax',
      'peace',
      'verve',
      'uplift',
      'deepPurple',
      'stamina',
      'deep',
      'medium',
      'light',
      'clear',
      'white',
    ]),
    'Use `color` system prop instead',
  ),
  /** (deprecated: use fontSize instead) style the text following the theme */
  size: deprecated(fontSizes, 'Use `fontSize` system prop instead'),
  /** set the font-weight to regular */
  light: bool,
  fontSize: fontSizes,
};

Text.defaultProps = {
  inverted: false,
  variant: undefined,
  size: undefined,
  fontSize: 'medium',
  numberOfLines: undefined,
  light: false,
};

export {
  TextRenderer,
  Text,
  H1,
  H2,
  H3,
  H4,
  H5,
  Small,
  Tiny,
  Light,
  Regular,
  Medium,
  Bold,
  Black,
  SectionTitle,
  SmallestException,
  Display1,
  Display2,
  Display3,
  Display4,
  DisplayNumber,
};
