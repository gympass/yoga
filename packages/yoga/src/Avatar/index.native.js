import Avatar from './native/Avatar';
import AvatarCircle from './native/AvatarCircle';

const ExportAvatar = Object.assign(Avatar, { Circle: AvatarCircle });

export default ExportAvatar;
