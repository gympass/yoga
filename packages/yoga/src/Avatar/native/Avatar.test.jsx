import React from 'react';
import { render } from '@testing-library/react-native';

import { UserFilled } from '@gympass/yoga-icons';
import { ThemeProvider } from '../..';
import Avatar from './Avatar';
import Text from '../../Text';

describe('<Avatar />', () => {
  it('should match snapshot in default avatar', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Avatar width={78} height={78} fill="black" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should match snapshot in default avatar with placeholder prop', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Avatar placeholder={UserFilled} />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should match snapshot in circle avatar', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Avatar.Circle />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should match snapshot in circle avatar with placeholder prop', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Avatar.Circle placeholder={UserFilled} />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should match snapshot in avatar with children prop', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Avatar>
          <Text>AL</Text>
        </Avatar>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
