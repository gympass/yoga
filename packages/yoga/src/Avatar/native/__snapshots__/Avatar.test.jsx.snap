// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should match snapshot in avatar with children prop 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="small"
  display="flex"
  elevation="small"
  height={48}
  justifyContent="center"
  overflow="hidden"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 8,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": 48,
      },
    ]
  }
  width={48}
>
  <Text
    inverted={false}
    light={false}
    size="medium"
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
      ]
    }
  >
    AL
  </Text>
</View>
`;

exports[`<Avatar /> should match snapshot in circle avatar 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="circle"
  display="flex"
  elevation="small"
  height={48}
  justifyContent="center"
  overflow="hidden"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 9999,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": 48,
      },
    ]
  }
  width={48}
>
  <RNSVGSvgView
    bbHeight="50%"
    bbWidth="50%"
    fill="#FFFFFF"
    focusable={false}
    height="50%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Array [
          Object {
            "height": "50%",
            "width": "50%",
          },
        ],
        Object {
          "opacity": 1,
        },
        Object {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          4294967295,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;

exports[`<Avatar /> should match snapshot in circle avatar with placeholder prop 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="circle"
  display="flex"
  elevation="small"
  height={48}
  justifyContent="center"
  overflow="hidden"
  placeholder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 9999,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": 48,
      },
    ]
  }
  width={48}
>
  <RNSVGSvgView
    bbHeight="50%"
    bbWidth="50%"
    fill="#FFFFFF"
    focusable={false}
    height="50%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Array [
          Object {
            "height": "50%",
            "width": "50%",
          },
        ],
        Object {
          "opacity": 1,
        },
        Object {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          4294967295,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;

exports[`<Avatar /> should match snapshot in default avatar 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="small"
  display="flex"
  elevation="small"
  height={78}
  justifyContent="center"
  overflow="hidden"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 8,
        "display": "flex",
        "height": 78,
        "justifyContent": "center",
        "overflow": "hidden",
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": 78,
      },
    ]
  }
  width={78}
>
  <RNSVGSvgView
    bbHeight="50%"
    bbWidth="50%"
    fill="black"
    focusable={false}
    height="50%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Array [
          Object {
            "height": "50%",
            "width": "50%",
          },
        ],
        Object {
          "opacity": 1,
        },
        Object {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          4278190080,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;

exports[`<Avatar /> should match snapshot in default avatar with placeholder prop 1`] = `
<View
  alignItems="center"
  bgColor="elements.selectionAndIcons"
  borderRadius="small"
  display="flex"
  elevation="small"
  height={48}
  justifyContent="center"
  overflow="hidden"
  placeholder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#9898A6",
        "borderRadius": 8,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": 48,
      },
    ]
  }
  width={48}
>
  <RNSVGSvgView
    bbHeight="50%"
    bbWidth="50%"
    fill="#FFFFFF"
    focusable={false}
    height="50%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Array [
          Object {
            "height": "50%",
            "width": "50%",
          },
        ],
        Object {
          "opacity": 1,
        },
        Object {
          "flex": 0,
          "height": "50%",
          "width": "50%",
        },
      ]
    }
    width="50%"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          4294967295,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;
